.github/workflows/CI.yml

```diff
@@ -7,161 +7,125 @@ on:
   pull_request: {}
 
 jobs:
-  test-workspace:
-
-    runs-on: ${{ matrix.os }}
-
+  check:
+    # Run `cargo check` first to ensure that the pushed code at least compiles.
+    runs-on: ubuntu-latest
     strategy:
       matrix:
-        os: [ubuntu-18.04, windows-2019, macOS-10.14]
-        rust: [stable, nightly]
-
+        rust: [stable, 1.39.0]
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v1
-      - name: Install rust toolchain
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: ${{ matrix.rust }}
-          override: true
-      - name: Build
-        uses: actions-rs/cargo@v1
-        with:
-          command: check
-          args: --verbose
-      - name: Run tests
-        uses: actions-rs/cargo@v1
-        env:
-          CI: 'True'
-          RUSTFLAGS: '-D warnings'
-        with:
-          command: test
-          args: --verbose
-
-  test:
-
+    - uses: actions/checkout@master
+    - uses: actions-rs/toolchain@v1
+      with:
+        toolchain: ${{ matrix.rust }}
+        profile: minimal
+    - name: Check
+      uses: actions-rs/cargo@v1
+      with:
+        command: check
+        args: --all --bins --examples --tests --benches --all-features
```

Comment:
Why does this explicitly enumerate instead of using, say, `--all-targets`?

Reply:
not totally sure will replace that